langchain4j-qianfan/src/main/java/dev/langchain4j/model/qianfan/InternalQianfanHelper.java

```diff
@@ -179,10 +179,18 @@ static String getSystemMessage(List<ChatMessage> messages) {
 
     }
     public static List<Message> toOpenAiMessages(List<ChatMessage> messages) {
-        return messages.stream()
+         List<Message> aiMessages = messages.stream()
                 .filter(chatMessage -> !(chatMessage instanceof SystemMessage))
                 .map(InternalQianfanHelper::toQianfanMessage)
                 .collect(toList());
+
+         // ensure the length of messages is odd
+        if(aiMessages.size() == 0 || aiMessages.size() % 2 == 1){
+            return aiMessages;
+        }
+
+        aiMessages.remove(0);
```

Comment:
worth adding a log on WARN level? that the message was removed

Reply:
sorry, I couldn't find a log demo in llm model module. May we proceed without it?